### PR TITLE
Add ppc64le builds to Travis

### PR DESCRIPTION
**Pull request by adamretter**
Let's see how this goes...

### CODE REVIEW

**Comment by adamretter** [comment]
With GCC 5.4.0 (which is the default) for PPC64le on Travis-CI, we seem to have a compilation problem with our version of xxHash. I sent a PR upstream to see if they have the same issue, but it does not appear - https://github.com/Cyan4973/xxHash/pull/283

Perhaps @pdillinger might have an idea, or even @Cyan4973 might be interested to take a look. I am no expert in this area, but having fiddled around a bit it seems like it could be a compiler bug, but it is strange that it doesn't appear also in the upstream xxHash PPC64le compilation on Travis-CI; Perhaps some cumulative compiler state thing?!?

**Comment by pdillinger** [comment]
(rebase + fix conflicts to see if compiler error still occurs)

File: .travis.yml [hunk 1]
```diff
@@ -80,10 +93,12 @@ install:
   - if [ "${JOB_NAME}" == cmake-mingw ]; then
       sudo apt-get install -y mingw-w64 ;
     fi
-  - if [[ "${JOB_NAME}" == cmake* ]] && [ "${TRAVIS_OS_NAME}" == linux ]; then
+  - if [[ "${JOB_NAME}" == cmake* ]] && [ "${TRAVIS_OS_NAME}" == linux ] && [ "$(uname -m)" == x86_64 ]; then
       mkdir cmake-dist && curl --silent --fail --show-error --location https://github.com/Kitware/CMake/releases/download/v3.14.5/cmake-3.14.5-Linux-x86_64.tar.gz | tar --strip-components=1 -C cmake-dist -xz && export PATH=$PWD/cmake-dist/bin:$PATH;
+    elif [[ "${JOB_NAME}" == cmake* ]] && [ "${TRAVIS_OS_NAME}" == linux ] && [ "$(uname -m)" == ppc64le ]; then
+      mkdir cmake-dist && curl --silent --fail --show-error --location https://github.com/Kitware/CMake/releases/download/v3.14.5/cmake-3.14.5.tar.gz | tar --strip-components=1 -C cmake-dist -xz && mkdir cmake-dist/build && pushd cmake-dist/build && ../bootstrap && make && popd && export PATH=$PWD/cmake-dist/build/bin:$PATH;
```
Comment (pdillinger) [inline .travis.yml]: Building from source looks to take almost 20 minutes. Can you put together a binary .tar.bz2 for this?

Comment (adamretter) [reply .travis.yml]: I already asked upstream, but no progress - https://gitlab.kitware.com/cmake/cmake/issues/20122

I could build and host our own?

Comment (adamretter) [reply .travis.yml]: @pdillinger here ya go, binaries for ppc64le - http://static.adamretter.org.uk/cmake-3.14.5-Linux-ppc64le.tar.bz2

Comment (pdillinger) [reply .travis.yml]: Hmm. Is that build missing a download capability? https://travis-ci.org/facebook/rocksdb/jobs/660214094?utm_medium=notification&utm_source=github_status

Comment (adamretter) [reply .travis.yml]: Ah! It looks like the cmake I build for ppc64le maybe doesn't support HTTPS (SSL) - let me build a new one...

Comment (adamretter) [reply .travis.yml]: @pdillinger I have uploaded a newer [cmake-3.14.5-Linux-ppc64le.tar.bz2]( http://static.adamretter.org.uk/cmake-3.14.5-Linux-ppc64le.tar.bz2), hopefully this one includes SSL support. The new versions md5 checksum is `5dd300852435fa9eca7d519098bed2a0`. 

Can you replace your copy and re-trigger the build?

Comment (pdillinger) [reply .travis.yml]: Done

**Comment by adamretter** [comment]
@pdillinger Yay all tests are green :-) Would you merge this, then we can rebase the s390x and aarch64 PRs atop this.

**Pull request closed without merging**